### PR TITLE
Read only mode

### PR DESCRIPTION
**Pull request by AlinaGoaga**
Closes [3513](https://app.zenhub.com/workspaces/team-pesto-6127680175876a0014fc5657/issues/gh/weaveworks/weave-gitops-enterprise/3513)

### CODE REVIEW

File: plugins/backstage-plugin-flux/config.d.ts [hunk 1]
```diff
@@ -2,5 +2,6 @@ export interface Config {
   gitops?: {
     /** @visibility frontend */
     baseUrl?: string;
+    readonly?: boolean;
```
Comment (foot) [inline plugins/backstage-plugin-flux/config.d.ts:5]:
```suggestion
    readOnly?: boolean;
```

File: plugins/backstage-plugin-flux/config.d.ts [hunk 2]
```diff
@@ -2,5 +2,6 @@ export interface Config {
   gitops?: {
     /** @visibility frontend */
     baseUrl?: string;
+    readOnly?: boolean;
```
Comment (foot) [inline plugins/backstage-plugin-flux/config.d.ts:5]: I think we need to set the visibility on this attr to frontend too?

Does it work if you

`kubectl apply -k demo`
`./startup.sh`

etc?

File: app-config.yaml [hunk 3]
```diff
@@ -109,6 +109,7 @@ catalog:
 
 gitops:
   baseUrl: ${WEAVE_GITOPS_URL}
+  #  readOnly: false # Set to true to disable sync/suspend/resume of resources
```
Comment (foot) [inline app-config.yaml:112]:
```suggestion
  # readOnly: true # Set to true to disable sync/suspend/resume of resources
```

Just to uncomment and go

File: plugins/backstage-plugin-flux/README.md [hunk 4]
```diff
@@ -282,9 +282,21 @@ kubernetes:
           caData: LS0tLS1CRUdJTiBDRVJUSUZJQ0...
 ```
 
+You can also update the configuration to restrict the access to the sync/suspend/resume of resources. To do this, simple set readOnly to true.
+
+```yaml
+# app-config.yaml
+
+gitops:
+  #
```
Comment (foot) [inline plugins/backstage-plugin-flux/README.md:291]:
```suggestion
```

File: plugins/backstage-plugin-flux/src/components/helpers.tsx [hunk 5]
```diff
@@ -146,10 +151,12 @@ export function ResumeButton({
...
   status,
+  readOnly,
 }: {
   resource: Source | Deployment;
   toggleResume: () => Promise<void>;
   status: boolean;
+  readOnly?: boolean;
 }) {
   const classes = useStyles();
   const label = `${resource.namespace}/${resource.name}`;
```
[the remark this answers is not in the document]
Comment (AlinaGoaga) [reply plugins/backstage-plugin-flux/src/components/helpers.tsx:162]: Good idea, yes!

File: plugins/backstage-plugin-flux/README.md [hunk 6]
```diff
@@ -282,9 +282,20 @@ kubernetes:
...
+You can also update the configuration to restrict the access to the sync/suspend/resume of resources. To do this, simple set readOnly to true.
+
+```yaml
+# app-config.yaml
+
+gitops:
+  readOnly: true
+```
+
+The UI will reflect this by showing the buttons corresponding to sync/suspend/resume as disabled.
```
Comment (foot) [inline plugins/backstage-plugin-flux/README.md:294]: We should probably call out here this is only to disable buttons in the UI which should be used as **part** of a readOnly solution which should also include disabling the k8s-proxy https://backstage.io/docs/features/kubernetes/proxy 

> [!IMPORTANT]
> Enabling `readOnly` mode disables the buttons in the UI and provides feedback to the user that resources cannot be changed. However this is only part of a full `readOnly` solution. The [Backstage Kubernetes Proxy](https://backstage.io/docs/features/kubernetes/proxy) should be configured to deny users from changing resources.

Kind of thing? @bigkevmcd

**Review by foot — APPROVED** [review]
But otherwise this is looking great!

:dancing-duck-in-a-hat:

File: plugins/backstage-plugin-flux/README.md [hunk 7]
```diff
@@ -282,9 +282,21 @@ kubernetes:
...
+Enabling readOnly mode disables the buttons in the UI and provides feedback to the user that resources cannot be changed. However this is only part of a full readOnly solution. The [Backstage Kubernetes Proxy](https://backstage.io/docs/features/kubernetes/proxy/) should be configured to deny users from changing resources.
```
Comment (foot) [inline plugins/backstage-plugin-flux/README.md:295]:
```suggestion
Enabling readOnly mode disables the buttons in the UI and provides feedback to the user that resources cannot be changed. However this is only part of a full readOnly solution. The [Backstage Kubernetes Proxy](https://backstage.io/docs/features/kubernetes/proxy/) should be configured to deny users from changing resources and/or the ServiceAccount that Backstage uses to access the cluster should be limited to only have read permissions.
```

**Comment by foot** [comment]
Cool beans 👍

**Pull request merged**